java/source/demo/pom.xml

```diff
@@ -41,7 +41,7 @@
     <properties>
         <maven.compiler.source>11</maven.compiler.source>
         <maven.compiler.target>11</maven.compiler.target>
-        <arrow.version>18.1.0</arrow.version>
+        <arrow.version>19.0.1</arrow.version>
```

Comment:
Seems this got messed up?

Reply:
Yeah, I forgot we needed to change the process to separate out Java. I put up a PR for that at https://github.com/apache/arrow-cookbook/pull/376, re-ran the script with the changes in that PR, and force-pushed to this branch.